Improve landing page

mrdrogdrog opened the pull request:
### Description
This PR
  - changes the behavior for links so they don't open new tabs
  - improves the title of the documentation header link
  - Rephrases the introduction section
  - Adds two new (for us obvious, but maybe not for others) core features (web-based and self hosted)

### Steps

<!-- Please tick all steps this PR performs (if something is not necessary, please remove it) -->

- [x] Added changes
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc.github.io/blob/main/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

davidmehren commented:
Tbh, I really don't like that there is no "get started" button above the fold anymore.
[All](https://nextcloud.com/) [these](https://rocket.chat/) [other](https://www.discourse.org/) [open](https://bigbluebutton.org/) [source](https://www.drupal.org/) [projects](https://typo3.org/) either have a big button on the main page or in the navbar.

ErikMichelson commented:
Especially when leaving out the CTA-Buttons, I'm unsure about the label "Download" in the navigation bar.

There is no hint that HedgeDoc is a self-hostable web application within the first screen height. As a user not knowing anything about HedgeDoc, I might try the download page and be irritated what I get there.

mrdrogdrog commented:
But this problem is also present if you have the CTA buttons. I guess we should rephrase the first section then.

mrdrogdrog commented:
I just noticed that it's not written anywhere on the landing page.

mrdrogdrog commented:
Okay. I changed the first two sections a bit. I added these points "HedgeDoc is self hosted" - "HedgeDoc is web based".
However I'm still not very happy with the text for the "self host" feature and actually I don't want to use the firefox icon for "web based". But I don't know what else to use. Any idea?

davidmehren commented on data/en/features.yml:
```diff
@@ -6,6 +6,14 @@
   columns:
+    - title: "Web-based"
+      icon: "fa fa-firefox"
+      content: "Your users don't need extra software. HedgeDoc runs in their browser."
```
```suggestion
      content: "You don't need extra software. HedgeDoc runs in the browser."
```

mrdrogdrog replied:
I also thought about this but that sounds a bit like "You don't need to install anything. Not even the software itself! Just use your browser and it magically works!". But on the other hand the next point is "HedgeDoc runs on your server." Sooo I'm not sure.

mrdrogdrog replied:
(I changed it for now)

mrdrogdrog replied:
How about "The editor runs in the browser" "The markdown editor runs in the browser" ?

mrdrogdrog replied:
Changed it to "The editor runs in the browser"

davidmehren commented on data/en/features.yml:
```diff
@@ -6,6 +6,14 @@
+      icon: "fa fa-firefox"
+      content: "Your users don't need extra software. HedgeDoc runs in their browser."
+
+    - title: "Self hosted"
```
```suggestion
    - title: "Self-hosted"
```

davidmehren commented on data/en/features.yml:
```diff
@@ -6,6 +6,14 @@
+
+    - title: "Self hosted"
+      icon: "fa fa-server"
+      content: "Keep the data where they belong. On your server."
```
```suggestion
      content: "Stay in control of your data. Keep it on your server."
```

mrdrogdrog commented:
About the icon: how about https://forkaweso.me/Fork-Awesome/icon/html5/ ?

ErikMichelson commented on data/en/features.yml:
```diff
@@ -6,6 +6,14 @@
   columns:
+    - title: "Web-based"
+      icon: "fa fa-html5"
+      content: "You don't need extra software. The editor runs in the browser."
```
Maybe "runs in your browser"?

(Avoiding repeating _the_)

mrdrogdrog replied:
Makes sense

mrdrogdrog commented on content/english/_index.html:
```diff
@@ -4,31 +4,30 @@
+You can use it to easily collaborate on notes, graphs and even presentations in real-time.
+All you need to do is to share your note-link to your co-workers and they're ready to go.
+
+{{< button external=true label="Read the install guide" link="https://docs.hedgedoc.org/setup/getting-started" >}}{{< button label="Try the demo" link="https://demo.hedgedoc.org/" outline=true >}}
```
fixed
